### ISSUE TITLE
[AppMeshBackend] Support virtual gateway + virtual gateway route methods

### ISSUE THREAD

**Issue by armichaud** [getmoto/moto#8002]
Enable mocking of the APIs governing ingress into an AppMesh _via_ virtual gateways. 

Although I do think that its best to wait on this until https://github.com/getmoto/moto/issues/8001 is fulfilled (it will probably borrow a lot from the work to enable virtual nodes and their routes), I've filed this as a separate issue since I suspect it's lower priority for people who are trying to test their AppMeshes.
